pxtblocks/blocklyloader.ts

```diff
@@ -540,48 +512,113 @@ namespace pxt.blocks {
-                    .appendField("do");
-                this.setPreviousStatement(true);
-                this.setNextStatement(true);
-                this.setTooltip(lf("Run the same sequence of actions while the condition is met."));
+                    .appendField(lf("_while_do")); // "do" in English
+
+		setHelpResources(this,
+		    'device_while',
+		    lf("a loop that repeats while the condition is true"),
+		    lf("Run the same sequence of actions while the condition is met."),
+		    '/blocks/loops/while'
+		);
             }
         };
 
+        // pxt controls_simple_for
         Blockly.Blocks['controls_simple_for'] = {
             /**
              * Block for 'for' loop.
              * @this Blockly.Block
              */
             init: function () {
-                this.setHelpUrl("/blocks/loops/for");
-                this.setColour((<any>Blockly.Blocks).loops.HUE);
-                this.appendDummyInput()
-                    .appendField("for")
-                    .appendField(new Blockly.FieldVariable(null), 'VAR')
-                    .appendField("from 0 to");
-                this.appendValueInput("TO")
-                    .setCheck("Number")
-                    .setAlign(Blockly.ALIGN_RIGHT);
+                this.jsonInit({
+                  "message0": lf("for %1 from 0 to %2"),
+                  "args0": [
+                    {
+                      "type": "field_variable",
+                      "name": "VAR",
+                      "variable": "item"
+                    },
+                    {
+                      "type": "input_value",
+                      "name": "TO",
+                      "check": "Number"
+                    }
+                  ],
+                  "previousStatement": null,
+                  "nextStatement": null,
+                  "colour": blockColors['loops'],
+                  "inputsInline": true
+                });
                 this.appendStatementInput('DO')
-                    .appendField(Blockly.Msg.CONTROLS_FOR_INPUT_DO);
-                this.setPreviousStatement(true);
-                this.setNextStatement(true);
-                this.setInputsInline(true);
-                // Assign 'this' to a variable for use in the tooltip closure below.
+                    .appendField(lf("_for_do")); // "do" in English
+
                 let thisBlock = this;
-                this.setTooltip(function () {
-                    return "Have the variable '%1' take on the values from 0 to the end number, counting by 1, and do the specified blocks.".replace('%1',
-                        thisBlock.getFieldValue('VAR'));
-                });
+		setHelpResources(this,
+		    'controls_simple_for',
+		    lf("a loop that repeats the number of times you say"),
+		    function () {
+			return lf("Have the variable '%1' take on the values from 0 to the end number, counting by 1, and do the specified blocks.").replace('%1', thisBlock.getFieldValue('VAR'));
```

Comment:
``lf`` supports arguments which will be substitued to {0}, {1}, ...

See example at https://github.com/Microsoft/pxt/blob/c3f64ddcd6533ff8338346d12a7e291b03c7a7f1/webapp/src/appcache.ts#L13